### PR TITLE
feature/tom-baxter-rps (note: this is one giant commit due to git tomfoolery)

### PR DESCRIPTION
**Pull request by thomas99baxter**

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Game
+  # had this in a switch statement but it
+  # was a bit big and a bit daft so moving to use a hash.
+  # would kind of like to move this out of this file though
+  SCORES = {
+    :rock => { :rock => :draw, :paper => :paper, :scissors => :rock },
+    :scissors => { :rock => :rock, :paper => :scissors, :scissors => :draw },
+    :paper => { :rock => :paper, :paper => :draw, :scissors => :scissors },
+  }.freeze
```
Comment (troysnowden) [inline lib/game.rb:9]: what does .freeze do?

File: lib/opponent.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Opponent
+
+  def computer_hand
+    ["Rock", "Paper", "Scissors"].sample
+  end
+
+end
```
Comment (troysnowden) [inline lib/opponent.rb:7]: Like the use of another class for opponent, something I should have implemented. Nice following of the S in SOLID.

File: spec/features/homepage_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  scenario 'have correct content' do
+    visit('/')
+    expect(page).to have_content('Rock, Paper, Scissors')
+    expect(page).to have_content('Your name:')
+    expect(page).to have_field('name')
+    
+    expect(page).to have_content('Choose your move:')
+    expect(page).to have_button('Rock')
+    expect(page).to have_button('Paper')
+    expect(page).to have_button('Scissors')
```
Comment (troysnowden) [inline spec/features/homepage_spec.rb:11]: Asserting on all the content is probably a good idea, I think I just asserted I was on the right page (i.e. the title was correct or whatever) but this is better.

File: spec/features/web_helpers.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+def enter_details_and_play
+  visit '/'
+  page.fill_in('name', with: 'Tom')
+  page.click_button('Paper')
+  expect(page).to have_current_path('/result')
```
Comment (troysnowden) [inline spec/features/web_helpers.rb:5]: Haven't seen 'have_current_path', nice.

File: app.rb [hunk 5]
```diff
@@ -1,7 +1,21 @@
...
+    erb :index
+  end
+
+  post '/result' do
+    # should this be lowercase or capitalize?
+    # if i dont capitalize, something breaks, reviewer pls
+    @name = params[:name].capitalize
+    @player_hand = params[:choice]
+    @computer_hand = Opponent.new.computer_hand
+    @winner = Game.new.winner(@player_hand.downcase.to_sym, @computer_hand.downcase.to_sym)
```
Comment (troysnowden) [inline app.rb:17]: Is instantiating a new Opponent Object and Game Object every time you play one round the best idea?

File: lib/game.rb (hunk 1 above)
Comment (thomas99baxter) [reply lib/game.rb:9]: no idea, rubocop

File: app.rb (hunk 5 above)
Comment (thomas99baxter) [reply app.rb:17]: nah, couldve used a single instance here for sure. will consider this for future

**Pull request closed without merging**